sources/org.osbuild.curl

```diff
@@ -175,13 +155,13 @@ def download(items, cache):
             # check if url needs rhsm secrets
             if url.get("secrets", {}).get("name") == "org.osbuild.rhsm":
                 # rhsm secrets only need to be retrieved once and can then be reused
-                if rhsm_secrets is None:
+                if subscriptions is None:
                     try:
-                        rhsm_secrets = get_rhsm_secrets()
+                        subscriptions = Subscriptions.from_host_system()
                     except RuntimeError as e:
                         json.dump({"error": e.args[0]}, sys.stdout)
                         return 1
-                url["secrets"] = rhsm_secrets
+                url["secrets"] = subscriptions.get_secrets(url.get("url"))
```

Comment:
This potentially can throw a `RuntimeException`. I think we would need to catch that too and convert it to an `{"error": ...}` return like in the `try:` block above. I guess just moving there should work.